In [1]:
import zipfile
import os

zip_file_path = '/content/titanicDataset.zip'
output_dir = '/content/'

# Create the output directory if it doesn't exist
os.makedirs(output_dir, exist_ok=True)

# Unzip the file
with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
    zip_ref.extractall(output_dir)

print(f"'{zip_file_path}' unzipped to '{output_dir}'")
print("Contents of the unzipped directory:")
print(os.listdir(output_dir))

'/content/titanicDataset.zip' unzipped to '/content/'
Contents of the unzipped directory:
['.config', 'Titanic-Dataset.csv', 'titanicDataset.zip', 'sample_data']


In [2]:
import pandas as pd

# Load the dataset
df = pd.read_csv('/content/Titanic-Dataset.csv')

# Display the first 5 rows of the DataFrame
print(df.head())

# Display basic information about the DataFrame
print(df.info())

   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
<c

In [3]:
# Check for missing values
print("Missing values per column:")
print(df.isnull().sum())

Missing values per column:
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


### Handling Missing Values

Based on the missing values check:
- **Age**: 177 missing values. Will impute with the median.
- **Cabin**: 687 missing values. Will drop this column due to a high percentage of missing data.
- **Embarked**: 2 missing values. Will impute with the mode.

In [4]:
# Fill missing 'Embarked' values with the mode
most_frequent_embarked = df['Embarked'].mode()[0]
df['Embarked'].fillna(most_frequent_embarked, inplace=True)

# Fill missing 'Age' values with the median
median_age = df['Age'].median()
df['Age'].fillna(median_age, inplace=True)

# Drop the 'Cabin' column due to many missing values
df.drop('Cabin', axis=1, inplace=True)

# Verify that missing values have been handled
print("Missing values after handling:")
print(df.isnull().sum())

Missing values after handling:
PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       0
dtype: int64


/tmp/ipykernel_2696/2035501183.py:3: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['Embarked'].fillna(most_frequent_embarked, inplace=True)
/tmp/ipykernel_2696/2035501183.py:7: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=T

In [5]:
# Display the first few rows of the DataFrame after handling missing values
print("DataFrame head after handling missing values:")
print(df.head())

DataFrame head after handling missing values:
   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Embarked  
0      0         A/5 21171   7.2500        S  
1      0          PC 17599  71.2833        C  
2      0  STON/O2. 3101282   7.9250        S  
3      0            113803  53.1000        S  
4      0            373450   8.0500    

### Encoding Categorical Features

Converting categorical features like 'Sex' and 'Embarked' into a numerical representation using one-hot encoding.

In [6]:
# Perform one-hot encoding for 'Sex' and 'Embarked' columns
df = pd.get_dummies(df, columns=['Sex', 'Embarked'], drop_first=True)

# Display the first few rows of the DataFrame after encoding
print("DataFrame head after one-hot encoding:")
print(df.head())

# Display the new columns and their types to confirm encoding
print("\nDataFrame info after encoding:")
print(df.info())

DataFrame head after one-hot encoding:
   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name   Age  SibSp  Parch  \
0                            Braund, Mr. Owen Harris  22.0      1      0   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  38.0      1      0   
2                             Heikkinen, Miss. Laina  26.0      0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  35.0      1      0   
4                           Allen, Mr. William Henry  35.0      0      0   

             Ticket     Fare  Sex_male  Embarked_Q  Embarked_S  
0         A/5 21171   7.2500      True       False        True  
1          PC 17599  71.2833     False       False       False  
2  STON/O2. 3101282   7.9250     False       False        True  
3            113803  53.100

### Re-checking Missing Values

Confirming that all missing values have been handled and no new ones were introduced during encoding.

In [7]:
# Check for missing values after one-hot encoding
print("Missing values per column after encoding:")
print(df.isnull().sum())

Missing values per column after encoding:
PassengerId    0
Survived       0
Pclass         0
Name           0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Sex_male       0
Embarked_Q     0
Embarked_S     0
dtype: int64


### Removing Duplicate Rows

In [8]:
# Check the number of rows before removing duplicates
print(f"DataFrame shape before removing duplicates: {df.shape}")

# Remove duplicate rows
df.drop_duplicates(inplace=True)

# Check the number of rows after removing duplicates
print(f"DataFrame shape after removing duplicates: {df.shape}")

DataFrame shape before removing duplicates: (891, 12)
DataFrame shape after removing duplicates: (891, 12)


### Converting Data Types

Converting boolean columns ('Sex_male', 'Embarked_Q', 'Embarked_S') to integer type (0 or 1).

In [9]:
# Convert boolean columns to integer type (0s and 1s)
df['Sex_male'] = df['Sex_male'].astype(int)
df['Embarked_Q'] = df['Embarked_Q'].astype(int)
df['Embarked_S'] = df['Embarked_S'].astype(int)

print("DataFrame head after data type conversion:")
print(df.head())

print("\nDataFrame info after data type conversion:")
print(df.info())

DataFrame head after data type conversion:
   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name   Age  SibSp  Parch  \
0                            Braund, Mr. Owen Harris  22.0      1      0   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  38.0      1      0   
2                             Heikkinen, Miss. Laina  26.0      0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  35.0      1      0   
4                           Allen, Mr. William Henry  35.0      0      0   

             Ticket     Fare  Sex_male  Embarked_Q  Embarked_S  
0         A/5 21171   7.2500         1           0           1  
1          PC 17599  71.2833         0           0           0  
2  STON/O2. 3101282   7.9250         0           0           1  
3            113803  53

### Renaming Columns

Renaming 'SibSp' to 'Siblings_Spouses', 'Parch' to 'Parents_Children', and 'Pclass' to 'Passenger_Class' for improved clarity.

In [10]:
# Rename columns for better readability
df.rename(columns={
    'SibSp': 'Siblings_Spouses',
    'Parch': 'Parents_Children',
    'Pclass': 'Passenger_Class'
}, inplace=True)

print("DataFrame head after renaming columns:")
print(df.head())

print("\nDataFrame info after renaming columns:")
print(df.info())

DataFrame head after renaming columns:
   PassengerId  Survived  Passenger_Class  \
0            1         0                3   
1            2         1                1   
2            3         1                3   
3            4         1                1   
4            5         0                3   

                                                Name   Age  Siblings_Spouses  \
0                            Braund, Mr. Owen Harris  22.0                 1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  38.0                 1   
2                             Heikkinen, Miss. Laina  26.0                 0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  35.0                 1   
4                           Allen, Mr. William Henry  35.0                 0   

   Parents_Children            Ticket     Fare  Sex_male  Embarked_Q  \
0                 0         A/5 21171   7.2500         1           0   
1                 0          PC 17599  71.2833         0         